db_processor.bal

```diff
@@ -154,8 +154,28 @@ function updatePledge(string status, int pledgeId) returns Pledge|error {
 }
 
 function deletePledge(int pledgeId) returns error? {
-    _ = check dbClient->execute(`DELETE FROM PLEDGE_UPDATE WHERE PLEDGEID=${pledgeId}`);
-    _ = check dbClient->execute(`DELETE FROM PLEDGE WHERE PLEDGEID=${pledgeId};`);
+    string message = "";
+    transaction {
+        sql:ExecutionResult|error deletePledgeUpdateResult = check dbClient->execute(
```

Comment:
Since there is check at RHS, it will never return error. So having `sql:ExecutionResult|error` is incorrect.

Reply:
Also better not to use two variables as `deletePledgeUpdateResult`  and `deletePledgeResult`. You can simply have `sql:ExecutionResult|error res` and move message generation logic next to that line and then reuse the same return variable to the next execution.

Reply:
Updated.